File: howManySundays.py
"""
We need a function that calculates the number of Sundays after a specific day for a given period.

Example

For n = 9 and startDay = "Saturday", the output should be
howManySundays(n, startDay) = 2;
For n = 7 and startDay = "Sunday", the output should be
howManySundays(n, startDay) = 1.
"""

def howManySundays(n, startDay):
    days = ['Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday', 'Sunday']
    
    cur_day = days.index(startDay)
    
    q = 0
    
    for i in range(n):
        cur_day += 1
        cur_day %= 7
        if cur_day == 6:
            q += 1
            
    return q

File: howManySundays.java
int howManySundays(int n, String startDay) {

  String[] week = {"Sunday", "Monday", "Tuesday", "Wednesday",
                   "Thursday", "Friday", "Saturday"};
  int startIndex = 0;

  for (int i = 0; i < week.length; i++) {
    if (week[i].equals(startDay)) {
      startIndex = i;
      break;
    }
  }

  return (int) Math.ceil((double)(n + startIndex) / week.length);
}
